### PR TITLE
Caliper section markings

### PR DESCRIPTION
**Pull request by fweik**
Start and stop profiler sections from python.

### CODE REVIEW

File: src/python/espressomd/profiler.pyx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def begin_section(name):
+    """
+    Start named section in profiler.
+
+    Parameters
+    ----------
+
+    name: obj:`str`
```
Comment (KaiSzuttor) [inline src/python/espressomd/profiler.pyx]: ```name : obj:`str` ```

File: src/python/espressomd/profiler.pyx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        Name of the section
+
+    """
+
```
Comment (KaiSzuttor) [inline src/python/espressomd/profiler.pyx:14]: remove blank

File: src/python/espressomd/profiler.pyx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    c_profiler.begin_section(name)
```
Comment (KaiSzuttor) [inline src/python/espressomd/profiler.pyx:15]: should we check if the profiler is available?
```
IF NOT HAVE_CALIPER:
    raise ValueError("No profiler available.")
```

Comment (fweik) [reply src/python/espressomd/profiler.pyx:15]: It does nothing if it's not available, which is fine.

File: src/python/espressomd/profiler.pyx (hunk 1 above)
Comment (fweik) [reply src/python/espressomd/profiler.pyx]: Does this affect the result?

Comment (KaiSzuttor) [reply src/python/espressomd/profiler.pyx]: idk

Comment (fweik) [reply src/python/espressomd/profiler.pyx]: Is this style documented somewhere? It does not produce a sphinx warning not to have the space

Comment (KaiSzuttor) [reply src/python/espressomd/profiler.pyx]: https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html

Comment (KaiSzuttor) [reply src/python/espressomd/profiler.pyx]: https://numpydoc.readthedocs.io/en/latest/format.html

Comment (fweik) [reply src/python/espressomd/profiler.pyx]: Can this be enforced automatically? Otherwise we should not require it.

Comment (KaiSzuttor) [reply src/python/espressomd/profiler.pyx]: I think no.

**Review by KaiSzuttor — APPROVED** [review]

**Pull request merged**